components/config/src/highlighting.rs

```diff
@@ -7,49 +8,60 @@ use syntect::parsing::SyntaxSet;
 use crate::config::Config;
 
 lazy_static! {
-    pub static ref SYNTAX_SET: SyntaxSet = {
+    pub static ref BUILTIN_SYNTAX_SET: SyntaxSet = {
         let ss: SyntaxSet =
             from_binary(include_bytes!("../../../sublime/syntaxes/newlines.packdump"));
         ss
     };
-    pub static ref THEME_SET: ThemeSet =
+    pub static ref BUILTIN_HIGHLIGHT_THEME_SET: ThemeSet =
         from_binary(include_bytes!("../../../sublime/themes/all.themedump"));
 }
 
-pub enum HighlightSource {
-    Theme,
+pub static EXTRA_SYNTAX_SET: OnceCell<SyntaxSet> = OnceCell::new();
+pub static EXTRA_HIGHLIGHT_THEME_SET: OnceCell<ThemeSet> = OnceCell::new();
```

Comment:
It is my understanding that you went for `OnceCell` because `ThemeSet` is not cloneable. I believe there is an easier way of resolving this problem: wrap it in an `Arc` instead.

Reply:
Hi, yes you are correct. I was working around clone. An Arc should work just as well, and without awkwardness around ensuring you only set it once.

Reply:
Ah, except `Arc<SyntaxSet>` doesn't impl Serialize/Deserialize. I don't know how to I would implement it manually, and I suspect it wouldn't be worth the hassle in this case?

Reply:
Done, sorted with skip_serialize/deserialize and `Arc<>`